weblab_cg/_chaste_printer.py

```diff
@@ -0,0 +1,91 @@
+from ._printer import WebLabPrinter
+import sympy
+
+from sympy.printing.precedence import precedence
+
+
+class ChastePrinter(WebLabPrinter):
+    """
+    Converts Sympy expressions to strings for use in Chaste code generation, based on the WeblabPrinter.
+
+    To use, create a :class:`ChastePrinter` instance, and call its method
+    :meth:`doprint()` with a Sympy expression argument.
+
+    Arguments:
+
+    ``symbol_function``
+        A function that converts symbols to strings (variable names).
+    ``derivative_function``
+        A function that converts derivatives to strings.
+
+    """
+
+    def __init__(self, symbol_function=None, derivative_function=None):
+        super().__init__(symbol_function, derivative_function)
+
+        # Do not use the math. prefix for functions, as chaste c++ files include an include for this
+        self._prefix = ''
+
+        # Make sure we can output a call to GetIntracellularAreaStimulus
+        WebLabPrinter._math_functions.add('GetIntracellularAreaStimulus')
+
+    def _print_And(self, expr):
+        """ Handles logical And. """
+        my_prec = precedence(expr)
+        return ' && '.join([self._bracket(x, my_prec) for x in expr.args])
+
+    def _print_BooleanFalse(self, expr):
+        """ Handles False """
+        return 'false'
+
+    def _print_BooleanTrue(self, expr):
+        """ Handles True """
+        return 'true'
+
+    def _print_Or(self, expr):
+        """ Handles logical Or. """
+        my_prec = precedence(expr)
+        return ' || '.join([self._bracket(x, my_prec) for x in expr.args])
+
+    def _print_Pow(self, expr):
+        """ Handles Pow(), which includes all division
+        only ordinary power is different, the rest is handed back up to the parent class (WebLabPrinter) """
+        # Square root or ((only if communicative) 1/sqrt or ordinary devision )
```

Comment:
I don't know what this comment means. Chances are you won't remember in a year either :D

Reply:
I've updated the docstring (will be in next commit)
        """ Handles Pow(), which includes all division
        
        for C++ printing we need to writea x**y as pow(x, y) with lowercase p
        only ordinary power is different from the weblab printer
        the rest (suqare root and division is handed back up to the parent class (WebLabPrinter) """

Reply:
Hhm it doesn't work without the brackets. Actually if you look at your own printer it also has:

`return self._bracket(expr.base, p) + '**' + self._bracket(expr.exp, p)`